components/filter/filter-dimension-set-date-time-range-value.js

```diff
@@ -0,0 +1,139 @@
+import '../inputs/input-date-time-range.js';
+import { html, LitElement } from 'lit';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { LocalizeCoreElement } from '../../helpers/localize-core-element.js';
+
+/**
+ * A component to represent a possible custom date range or date-time range value that can be selected for a dimension set (the main filter dimension type).
+ * The start-value and end-value will be included in the d2l-filter-change event and formatted as an ISO string in UTC time.
+ * This component does not render anything, but instead gathers data needed for the d2l-filter.
+ */
+class FilterDimensionSetDateTimeRangeValue extends LocalizeCoreElement(LitElement) {
+
+	static get properties() {
+		return {
+			/**
+			 * Whether this value in the filter is disabled or not
+			 * @type {boolean}
+			 */
+			disabled: { type: Boolean, reflect: true },
+			/**
+			 * @ignore
+			 */
+			endValue: { type: String, attribute: 'end-value' },
+			/**
+			 * REQUIRED: Unique key to represent this value in the dimension
+			 * @type {string}
+			 */
+			key: { type: String },
+			/**
+			 * Whether this value in the filter is selected or not
+			 * @type {boolean}
+			 */
+			selected: { type: Boolean, reflect: true },
+			/**
+			 * @ignore
+			 */
+			startValue: { type: String, attribute: 'start-value' },
+			/**
+			 * Defaults to "Custom Date Range" (localized). Can be overridden if desired.
+			 * @type {string}
+			 */
+			text: { type: String, reflect: true }
+		};
+	}
+
+	constructor() {
+		super();
+		this.disabled = false;
+		this.selected = false;
+		this._dispatchFilterChangeEvent = false;
+		this._enforceSingleSelection = true;
+		this._minWidth = 375;
+		this._filterSetValue = true;
+		this._noSearchSupport = true;
+
+		this._handleDateChange = this._handleDateChange.bind(this);
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+
+		this.text = this.text || this.localize('components.filter-dimension-set-date-time-range-value.text');
+	}
+
+	updated(changedProperties) {
+		super.updated(changedProperties);
+
+		const changes = new Map();
+		let shouldDispatchChangeEvent = false;
+		changedProperties.forEach((oldValue, prop) => {
+			if (oldValue === undefined && (prop === 'selected' || prop === 'disabled')) return;
+			if (this._dispatchFilterChangeEvent && (prop === 'startValue' || prop === 'endValue')) shouldDispatchChangeEvent = true;
+
+			if (prop === 'disabled' || prop === 'selected' || prop === 'startValue' || prop === 'endValue' || prop === 'text') {
+				changes.set(prop, this[prop]);
+			}
+		});
+		if (changes.size > 0) {
+			/** @ignore */
+			this.dispatchEvent(new CustomEvent('d2l-filter-dimension-set-value-data-change', {
+				detail: { valueKey: this.key, changes: changes, dispatchChangeEvent: shouldDispatchChangeEvent },
+				bubbles: true,
+				composed: false
+			}));
+			this._dispatchFilterChangeEvent = false;
```

Comment:
Generally the same as the other -value components with the addition of dispatchChangeEvent for start and end value